core/src/validator.rs

```diff
@@ -891,6 +892,31 @@ impl Validator {
             ip_echo_server.shutdown_background();
         }
     }
+
+    fn initiate_cost_model(cost_model: &RwLock<CostModel>, cost_table: &[(Pubkey, u64)]) {
+        let mut cost_model_mutable = cost_model.write().unwrap();
+        for (program_id, cost) in cost_table {
+            match cost_model_mutable.upsert_instruction_cost(program_id, cost) {
+                Ok(c) => {
+                    debug!(
+                        "initiating cost table, instruction {:?} has cost {}",
+                        program_id, c
+                    );
+                }
+                Err(err) => {
+                    debug!(
+                        "initiating cost table, failed for instruction {:?}, err: {}",
+                        program_id, err
+                    );
+                }
+            }
+        }
+        drop(cost_model_mutable);
+        debug!(
+            "restored cost model instruction cost table from blockstore, current values: {:?}",
+            cost_model.read().unwrap().get_instruction_cost_table()
+        );
```

Comment:
Believe this to be the case, but a failure to `upsert()` is a non-fatal error and we should just continue (as you're doing), correct?

Reply:
Correct - the best effort to furnish cost table during validator startup. But if not successful, table will be updated in runtime by replay_stage, not fatal event.